packages/web/modals/wallet-select.tsx

```diff
@@ -373,7 +373,55 @@ const ModalContent: FunctionComponent<
 
     const wallets = [...WalletRegistry]
       // If mobile, filter out browser wallets
-      .filter((w) => (isMobile ? !w.mobileDisabled : true))
+      .reduce((acc, w, _index, array) => {
+        if (isMobile) {
+          /**
+           * If an extension wallet is found in mobile, this means that we are inside an app browser.
+           * Therefore, we should only show that compatible extension wallet.
+           * */
+          if (acc.length > 0 && acc[0].name.endsWith("-extension")) {
+            return acc;
+          }
+
+          const windowAsRecord = window as Record<string, any>;
+          const mobileWebModeName = "mobile-web";
+
+          /**
+           * If on mobile and `leap` is in `window`, it means that the user enters
+           * the frontend from Leap's app in app browser. So, there is no need
+           * to use wallet connect, as it resembles the extension's usage.
+           */
+          if (
+            windowAsRecord?.leap &&
+            windowAsRecord?.leap?.mode === mobileWebModeName
+          ) {
+            return array
+              .filter((w) => w.name === "leap-extension")
```

Comment:
we should probably keep these strings e.g. 'leap-extension' 'keplr-extension' etc in constants somewhere

Reply:
Nice suggestion; I added an enum in my latest commit to search for the available wallet